src/openrct2/ride/track_design_save.c

```diff
@@ -1006,8 +1006,10 @@ static bool track_design_save_to_td6_for_tracked_ride(uint8 rideIndex, rct_track
     sint16 start_z = z + trackCoordinates->z_begin;
     gTrackPreviewOrigin = (LocationXYZ16) { start_x, start_y, start_z };
 
+    const uint16 maxTrackElements = 8192;
```

Comment:
This should be declared outside the function.

e.g:
`#define TD6_MAX_TRACK_ELEMENTS 8192`